File: algorithms/sorting/insertsort/insertsort.py
#!/usr/bin/python
# Copyright (C) 2014 Miquel Sabaté Solà <mikisabate@gmail.com>
# This file is licensed under the MIT license.
# See the LICENSE file.

def insertsort(lst):
    for i in range(1, len(lst)):
        value = lst[i]
        j = i
        while j > 0 and value < lst[j - 1]:
            lst[j] = lst[j - 1]
            j -= 1
        lst[j] = value


ary = [4, 65, 2, -31, 0, 99, 2, 83, 782, 1]
print(ary)

insertsort(ary)
print(ary)

File: algorithms/sorting/insertsort/insertsort.c
/*
 * Copyright (C) 2014 Miquel Sabaté Solà <mikisabate@gmail.com>
 * This file is licensed under the MIT license.
 * See the LICENSE file.
 */

#include <stdio.h>

void insertsort(int *v, int size)
{
	int i, j, value;

	for (i = 0; i < size; i++) {
		value = v[i];
		for (j = i; j > 0 && value < v[j - 1]; j--) {
			v[j] = v[j - 1];
		}
		v[j] = value;
	}
}

int main(int argc, char *argv[])
{
	int i;
	int v[] = {4, 65, 2, -31, 0, 99, 2, 83, 782, 1};
	int size = sizeof(v) / sizeof(v[0]);

	for (i = 0; i < size; i++) {
		printf("%i ", v[i]);
	}
	printf("\n");

	insertsort(v, size);

	for (i = 0; i < size; i++) {
		printf("%i ", v[i]);
	}
	printf("\n");
	return 0;
}
